libvast/src/plugin.cpp

```diff
@@ -380,6 +382,55 @@ store_plugin::make_store(system::accountant_actor accountant,
                                                  std::move(path), name());
 }
 
+// -- loader plugin -------------------------------------------------------------
+
+auto stdin_loader_plugin::make_loader(options, const bool_operator*) const
+  -> caf::expected<input_loader> {
+  return []() -> generator<chunk_ptr> {
+    auto in_buf = detail::fdinbuf(STDIN_FILENO, max_chunk_size);
```

Comment:
IIUC code using an rdbuf directly must be guarded with a `std::istream::sentry`, but I suppose that's irrelevant here because we don't actually use an istream but rather just the rdbuf? Is it a deliberate decision not to go through an istream for reading?

Reply:
The only reason why you'd go through an istream is when you want formatted input. Since we're doing block-level access of the file descriptor, that strikes me as the right (implicit?) design choice.

Reply:
`istream` has the ability to format and interpret in/output, which seems unnecessary in this case.